### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by andrii1**

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const documents = require("./documents.json");
+
+// Support parsing JSON requests
+app.use(express.json());
+
+//GET /search
+app.get("/search", (req, res) => {
+  if (req.query.q) {
+    const containsQuery = documents.filter(function (entry) {
+      return entry.description.includes(req.query.q);
```
Comment (orhantoy) [inline nodejs/week2/app.js]: This only searches the `description` field but the idea is to search all fields, at least the fields that contain strings.

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+});
+
+//POST /search
+app.post("/search", (req, res) => {
+  const filterBody = req.body.fields;
+
+  if (req.query.q && req.body.fields) {
+    res.status(404).json({ error: "Bad request" });
+  } else if (filterBody) {
+    const filterByType = req.body.fields.type;
```
Comment (orhantoy) [inline nodejs/week2/app.js]: This only supports a field called `type` but the idea was to support any field name.

**Review by orhantoy — CHANGES_REQUESTED** [review]
This is a good start but certain functionality is missing. You can see in https://swiftsearch.vercel.app/ what the expected functionality is. You are allowed to make simplifications though and only consider string fields.

File: nodejs/week2/app.js [hunk 3]
```diff
@@ -28,6 +28,23 @@ app.get("/documents/:id", async (req, res) => {
...
+    const filterResult = documents.filter(function (entry) {
+      return entry.type === filterByType;
+    });
+    res.send(filterResult);
+  } else {
+    res.status(404).json({ error: "Bad request, request body can't be empty" });
+  }
```
Comment (27ManishaPatel) [inline nodejs/week2/app.js:55]: great

**Review by 27ManishaPatel — APPROVED** [review]
Good work !

File: nodejs/week2/app.js (hunk 2 above)
Comment (andrii1) [reply nodejs/week2/app.js]: added this `app.post("/search", (req, res) => {
  if (req.query.q && req.body.fields) {
    res.status(404).json({ error: "Bad request" });
  } else if (req.body.fields.type) {
    const filter = req.body.fields.type;
    const filterResults = documents.filter(function (entry) {
      return entry.type === filter;
    });
    res.send(filterResults);
  } else if (req.body.fields.description) {
    const filter = req.body.fields.description;
    const filterResults = documents.filter(function (entry) {
      return entry.type === filter;
    });
    res.send(filterResults);
  } else {
    res.status(404).json({ error: "Bad request, request body can't be empty" });
  }
});`

**Pull request merged**